tensordict/tensordict.py

```diff
@@ -1952,12 +1952,16 @@ def __getitem__(self, idx: INDEX_TYPING) -> TensorDictBase:
             idx = torch.tensor(idx, device=self.device)
             return self._index_tensordict(idx)
 
+        if isinstance(idx, range):
+            idx = torch.tensor(idx, device=self.device)
+            return self._index_tensordict(idx)
```

Comment:
There's some repetition here, but I figured it's deliberate to benefit from early return in the most common case of indexing with a list. I also assumed that `range` would be less common than `np.ndarray`.

As a side-note: this seems to be the only place we handle `np.ndarray`, it's not considered in other `__setitem__` or `__getitem__` definitions.